### DISCUSSION TITLE
XPath - relative paths applied to parent element not actual element

### DISCUSSION THREAD

**Discussion by v-mwalk** [webdriverio/webdriverio#8046] [Q&A] [answered]
This is a bit odd and I need to drill down a lot more to fathom out what is happening.  However, I wanted to socialise it incase anyone else has this hit this oddness...  I'm also not sure, yet, where this - _potential_ - bug lies (webdriverio/appium/UIAutomator2 etc...)

I'm using webdriverio/appium against an Android native app.

const allMatchingElements = $$('//android.widget.SomeElement[@resource-id="xyzzy"]');

// Assume a single element matches - I have code that checks for that to ensure
// singleton (I'm making sure selector does not match multiple elements)....
const childMatchingElements = allMatchingElements[0].$$('./android.widget.ChildElement[@resource-id="yzzyx"]');

Although the xpaths are correct the second $$ locator does not work.  On investigation, basically trying and playing with the xpaths, I found it is because the './android.widget.ChildElement.......' is being applied to the PARENT element of allMatchingElements[0]

Currently, I'm not sure if allMatchingElements[0] IS the parent of who i think it is.  Or, if allMatchingElements[0] is correct, but the $$ is being applied to it's parent.  I am going to investigate further but I'm just on a tight timeline at the mo and so am using workarounds.....

I know that sounds weird, but it really is happening - as far as I can see.

Does anyone know where the code is I should be looking at (webdriverio?  appium?  UIAutomator2?  etc....)?

**Comment by kitoymonster** [comment]
Have you tried manually confirming the element location `//android.widget.SomeElement[@resource-id="xyzzy"]/android.widget.ChildElement[@resource-id="yzzyx"]` in appium inspector and see if that is a valid path?

**Reply by v-mwalk** [reply]
Yes.  However, it looks like it is actually an Appium issue.
https://github.com/appium/appium/discussions/16513

**Comment by v-mwalk** [comment, marked as answer]
Was an Appium issue.  Needs Appium 2 and setting limitXPathContextScope setting to false.

**Reply by robto09** [reply]
@v-mwalk Where this  limitXPathContextScope must be defined? Looks inside capabilities its not recognized, in webdriverio
